diff --git a/python/ambassador_diag/diagd.py b/python/ambassador_diag/diagd.py
index 3d9f8611a..cfec1add0 100644
--- a/python/ambassador_diag/diagd.py
+++ b/python/ambassador_diag/diagd.py
@@ -1800,8 +1800,16 @@ class AmbassadorEventWatcher(threading.Thread):
         self.app.config_timer.stop()
 
         if app.kick:
-            self.logger.debug("running '%s'" % app.kick)
-            os.system(app.kick)
+            match = re.match(r"kill -HUP ([0-9]+)", app.kick)
+            if match:
+                pid = int(match.group(1))
+                self.logger.debug("doing os.kill(%d, signal.SIGHUP) instead of running '%s'" % (pid, app.kick))
+                os.kill(pid, signal.SIGHUP)
+            else:
+                self.logger.warning("Trying to run '%s'" % app.kick)
+                self.logger.warning("You will need to install a shell via DHI customization for this to work.")
+                self.logger.debug("running '%s'" % app.kick)
+                os.system(app.kick)
         elif app.ambex_pid != 0:
             self.logger.debug("notifying PID %d ambex" % app.ambex_pid)
             os.kill(app.ambex_pid, signal.SIGHUP)
