diff --git a/tensorflow_serving/workspace.bzl b/tensorflow_serving/workspace.bzl
index c7b7a896..a2fbaa56 100644
--- a/tensorflow_serving/workspace.bzl
+++ b/tensorflow_serving/workspace.bzl
@@ -31,6 +31,8 @@ def tf_serving_workspace():
         sha256 = "8e00c38829d6785a2dfb951bb87c6974fa07dfe488aa5b25deec4b8bc0f6a3ab",
         strip_prefix = "rapidjson-1.1.0",
         build_file = "@//third_party/rapidjson:BUILD",
+        patches = ["@//third_party/rapidjson:3b2441b87f99ab65f37b141a7b548ebadb607b96.patch"],
+        patch_args = ["-p1"],
     )
 
     # ===== libevent (libevent.org) dependency =====
