diff --git a/project.clj b/project.clj
--- a/project.clj
+++ b/project.clj
@@ -1,7 +1,7 @@
 ;; This is Leiningen's own project configuration. See doc/TUTORIAL.md
 ;; file as well as sample.project.clj for help writing your own.
 
-(defproject leiningen "2.12.1-SNAPSHOT"
+(defproject leiningen "2.13.0"
   :description "Automate Clojure projects without setting your hair on fire."
   :url "https://codeberg.org/leiningen/leiningen"
   :license {:name "Eclipse Public License"
@@ -23,12 +23,13 @@
                  ;; needed for change
                  [org.clojars.trptcolin/sjacket "0.1.1.1"
                   :exclusions [org.clojure/clojure]]
                  ;; bump versions of various common transitive deps
                  [net.cgrand/parsley "0.9.3" :exclusions [org.clojure/clojure]]
                  [scout "0.1.1"]
-                 [commons-io "2.8.0"]
-                 [commons-codec "1.21.0"]]
-  :pedantic? :abort
+                 [commons-io "2.17.0"]
+                 [commons-codec "1.21.0"]
+                 [org.codehaus.plexus/plexus-utils "3.6.1"]]
+  :pedantic? :warn
   ;; checkout-deps don't work with :eval-in :leiningen
   :profiles {:dev {:resource-paths ["leiningen-core/dev-resources"]
                    :test-paths ["leiningen-core/test"]}
