diff --git a/gradle/verification-keyring.keys b/gradle/verification-keyring.keys
index b917a576ff1..a91f8c1e022 100644
--- a/gradle/verification-keyring.keys
+++ b/gradle/verification-keyring.keys
@@ -2946,6 +2946,47 @@ b2RkKSA8aW5mb0Bqb2RkLm9yZz4=
 =B5Xk
 -----END PGP PUBLIC KEY BLOCK-----
 
+pub    CA80D1F0EB6CA4BA
+sub    5EAB8AD72016DF52
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mQINBF62njsBEADakbaGRfpiftmwO/KncA+vG8cNJzPNEU1HD+o0ReMPO6H5G45b
+7gxhZut1Ag1jT/vPbSsTtCloCCy8WF7GtWbjCvvRd8SMP0dlH3vG3rnprXxUCnN4
+jxBCy5Gd/5uPy1G/pMgOwZ03Aam4QP6fQWQ1P+pKT36ZwYcl7jrriYT3jixDIpl3
+kTB9zEwdfN7YBqBczxk+xSlrPMYuYV9i1/+bMgWcRB7w2d+7dwrnxzx3RuUcCxKS
+bPQO4nrq9BalYbaJiGmc/M8sqlLnEsXZjGpFWH06OGSAvW+umyWZ8+nAueSJ3+2d
+6GrGf7lok7UL8zBN8h6TcwTboHY4VUiYk9Fx6Ep6gGxujD9B2hbsL1PWXAX/BqYx
+XE1PGNSB2AoZR4376qxVJnp/SphH7OiNsFOHpiQrrNKu/HOcZZlPLO01Fbkzgx0O
+T5nHRHaWhUvmqGRphogmUVEkNwXMO92FILLkcvBGlNmyGDesw5300QEijdcXLZzi
+LZBuyANzn7Ve0Bk8b6nNVaNln3yCkwLfrsROPItcj/rlEhS3bjvM5E+VmqBD6pvS
+kldJPDKJJGfJZFu7jx5x7kygrVZyFayJsSslnBGsyMvmORNhA1zzWR7tn1BNhCB1
+8JSiXAZEIrh6YOkK9mDlCYppoPZCUshCZVPD8l18eK7A2MQ8luQvgqC6dQARAQAB
+uQINBF62njsBEAC6cz5EQSsWCQHMPtqpMpLLD+B2iV1fnQjN7/i7X/cc3y50SwQ7
+PYJwbYFQ0zOtrTMIhL42B4JCgO7UGRlbGS0cV1U28XAeZ8QJMWgRcSMnye5XJzNV
+iO8Xsh9Z+8eFA0yhQWMrGL58DgqJrAWxW0JSyoNtqrOHceqamw63BMQGoA60eN3+
+4QLRhL0LTpOdSxCDqba+nXWivDxoigThr/fHqmL5fIJMhOdGrVhGH9l1YDLJBwLB
+EkZfwv4tcwSaJyGTBbSvukRqV/G0O6d3EQ98uS5zcMU96YrMxvS5CYoI/+G2zwzM
+inC54v68bWQTA2PoA94+tpSr3QJm7dA8aJv4/sia0v7lmx4hOabG62aKKuIDHbIF
+VQ5awhRt1mJ1YeerII0TigF6uIzdbqqyotxcZ9Abu926Y6P/ANMSV+BKhNNzqiRk
+VcqvrsTrwp3WZHlZCtSuAr8+dpBx0WEVLIhiilpXkyOm1oLw45idsYdZRPdhccLX
+J7ejyXDPZ6w60qRNs80LdCE+zVjGIPb6zZt0ZiqpEMeMOUDTpJfHBlWdN3irgGiG
+AYKKYk+RsxxhP35ny7Ps986IMJZcPRY7ElmUFjOwySYPykmDahiWXXZm1zPWTEIA
+Z6zBGHPaqc5ph2hiXDW8iMwLlgEF6YJb9cj4jSR4Xwgjtb/sKy0z28ecPQARAQAB
+iQI2BBgBCgAgFiEEMhGM92yexdkY5UlnyoDR8OtspLoFAl62njsCGwwACgkQyoDR
+8OtspLob3xAAmlXlX22kSOy6uV86MRHnRXwArPYO+sZYWtr3qwHJjo8O+HXvd0/+
+5b89M+rSUDEIoBkvlLAAVnp59SdFP4Cmau7KlIU+arKL/yx7/YC0fIqru2H8+pRZ
+3s7Hz6lq/EHv13y2jtabCA+f4QMeiiev+dkVUxODkLcryMZe4aZFfA7H25Kgd7h2
+X/+7x+57eNaZD6l9MGDxtxSM0To8cNdX9ivTIOupdTdTwiQ/gH6VeznC/9nhXCBE
+wiOKdcZcm+2tsGII9sYKR/EiehXOXTsKuD3nN8o1pFewqlhNzoqzaP7QhwpPV1w6
+AYClxZYcRKLgA8YwexW7g29R0dqhvxukwSsE0h2ftJl61FdfZI3zDs+A7UYat+B+
+l8SRtleFzRN9f8W8PUD1fev2r5QfMDEMsaJN0Uej+O3BNVyMC956aN3h9B2yKBJY
+W8xpi6J/ciB3S32UEDaOVejrxdk/yCsyZOUo11yVpfE+6OhiuBWCnuCdJRlPbrRd
+l2i3rgnmnql05cUUdosyNG8umx/U0hVWiuA3VPvOcb1y6r8gjVkO5G6WImoKV864
+NHpQ5L5JLK9ITyOLdB0M8ODvCZCaCNtvEkLfSsc/jUu6eMy3cMsHNomgTrLN/mmh
+do67iARWvgjnaggbagtEb4dhoX7YR7SKTRa9y77u8unIz+2IIM10Nw4=
+=7Ldz
+-----END PGP PUBLIC KEY BLOCK-----
+
 pub    CA85FFE638D4407A
 uid    Sebastiano Vigna <sebastiano.vigna@gmail.com>
 
diff --git a/gradle/verification-metadata.xml b/gradle/verification-metadata.xml
index 291b14c117c..37c48f35321 100644
--- a/gradle/verification-metadata.xml
+++ b/gradle/verification-metadata.xml
@@ -1308,6 +1308,11 @@
             <pgp value="F3D15B8FF9902805DE4BE6B18DC6F3D0ABDBD017"/>
          </artifact>
       </component>
+      <component group="org.codehaus.plexus" name="plexus-utils" version="3.6.1">
+         <artifact name="plexus-utils-3.6.1.jar">
+            <pgp value="32118CF76C9EC5D918E54967CA80D1F0EB6CA4BA"/>
+         </artifact>
+      </component>
       <component group="org.codehaus.woodstox" name="stax2-api" version="4.2.1">
          <artifact name="stax2-api-4.2.1.jar">
             <pgp value="6214760097DC5CFAD0175AC2C9FBAA83A8753994"/>
diff --git a/packaging/distributions-dependencies/build.gradle.kts b/packaging/distributions-dependencies/build.gradle.kts
index 1b2ea7e97c0..0afb173e598 100644
--- a/packaging/distributions-dependencies/build.gradle.kts
+++ b/packaging/distributions-dependencies/build.gradle.kts
@@ -152,7 +152,7 @@ dependencies {
         api(libs.plexusInterpolation)   { version { strictly("1.26"); because("transitive dependency of Maven modules to process POM metadata") }}
         api(libs.plexusClassworlds)     { version { strictly("2.7.0"); because("transitive dependency of Maven modules to process POM metadata") }}
         api(libs.plexusSecDispatcher)   { version { strictly("2.0"); because("transitive dependency of Maven modules to process POM metadata") }}
-        api(libs.plexusUtils)           { version { strictly("3.5.1"); because("transitive dependency of Maven modules to process POM metadata") }}
+        api(libs.plexusUtils)           { version { strictly("3.6.1"); because("transitive dependency of Maven modules to process POM metadata") }}
         api(libs.plist)                 { version { strictly("1.27") }}
         api(libs.servletApi)            { version { strictly("3.1.0") }}
         api(libs.slf4jApi)              { version { strictly(slf4jVersion) }}
