diff --git a/storage/src/main/java/org/apache/kafka/storage/internals/log/CleanerConfig.java b/storage/src/main/java/org/apache/kafka/storage/internals/log/CleanerConfig.java
index 4ae7779f5e..d3715a251f 100644
--- a/storage/src/main/java/org/apache/kafka/storage/internals/log/CleanerConfig.java
+++ b/storage/src/main/java/org/apache/kafka/storage/internals/log/CleanerConfig.java
@@ -34,7 +34,7 @@
  * Configuration parameters for the log cleaner.
  */
 public class CleanerConfig {
-    public static final String HASH_ALGORITHM = "MD5";
+    public static final String HASH_ALGORITHM = "SHA-256";
     public static final int LOG_CLEANER_THREADS = 1;
     public static final double LOG_CLEANER_IO_MAX_BYTES_PER_SECOND = Double.MAX_VALUE;
     public static final long LOG_CLEANER_DEDUPE_BUFFER_SIZE = 128 * 1024 * 1024L;
