diff --git i/docker/scripts/sh/entrypoint.sh w/docker/scripts/sh/entrypoint.sh
index edb2a1e22..1c2337165 100644
--- i/docker/scripts/sh/entrypoint.sh
+++ w/docker/scripts/sh/entrypoint.sh
@@ -1 +1 @@
-#!/bin/sh
+#!/bin/bash
@@ -7 +7 @@ set -eu -o pipefail
-: "${BIND_ON_IP:=$(getent hosts "$(hostname)" | awk '{print $1;}')}"
+: "${BIND_ON_IP:=$(cat /etc/hosts | grep -v ^# | grep "$HOSTNAME" | awk '{print $1;}')}"
@@ -12 +12 @@ if [ "${BIND_ON_IP}" = "0.0.0.0" ] || [ "${BIND_ON_IP}" = "::0" ]; then
-    : "${TEMPORAL_BROADCAST_ADDRESS:=$(getent hosts "$(hostname)" | awk '{print $1;}')}"
+    : "${TEMPORAL_BROADCAST_ADDRESS:=$(cat /etc/hosts | grep -v ^# | grep "$HOSTNAME" | awk '{print $1;}')}"
