diff --git a/servers/quarkus-server/build.gradle.kts b/servers/quarkus-server/build.gradle.kts
index 538479a57..92bb422ec 100644
--- a/servers/quarkus-server/build.gradle.kts
+++ b/servers/quarkus-server/build.gradle.kts
@@ -77,6 +77,15 @@ configurations.configureEach {
   }
 }
 
+configurations.configureEach {
+  resolutionStrategy {
+    force(
+      "io.quarkus:quarkus-vertx-http:3.35.2",
+      "org.postgresql:postgresql:42.7.11",
+    )
+  }
+}
+
 val quarkusRunner by
   configurations.creating {
     description = "Used to reference the generated runner-jar (either fast-jar or uber-jar)"
